keras_cv/layers/preprocessing/base_image_augmentation_layer.py

```diff
@@ -0,0 +1,288 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from keras.engine import base_layer
+from keras.layers.preprocessing import preprocessing_utils as utils
+from tensorflow.tools.docs import doc_controls
+
+H_AXIS = -3
+W_AXIS = -2
+
+IMAGES = "images"
+LABELS = "labels"
+TARGETS = "targets"
+BOUNDING_BOXES = "bounding_boxes"
+
+
+class BaseImageAugmentationLayer(base_layer.BaseRandomLayer):
```

Comment:
so you'll use the import that comes from

```
@keras_export("keras.__internal__.layers.BaseRandomLayer")
```

so,

tf.keras.__internal__.layers.BaseRandomLayer

Reply:
tf.keras.internals is not recognized - `AttributeError: module 'tensorflow.keras' has no attribute 'internal'`

Reply:
Were you using __internal__?

Reply:
you may need to upgrade tensorflow to 2.9.0

Reply:
yes `tf.keras.internal.layers.BaseRandomLayer`

Reply:
got it!